### PR TITLE
Updated as per dbt-teradata 1.7.1

### PR DESCRIPTION
**Pull request by tallamohan**
## What are you changing in this pull request and why?

- Updated compatibility matrix with dbt-core 1.7.4 support
- Model contracts are now supported with dbt-teradata v1.7.1 and onwards
- Addition of Tera Mode in dbt-teradata

### CODE REVIEW

**Comment by tallamohan** [comment]
Team,

Can you please review is PR?

Thanks,
Mohan

File: website/docs/docs/core/connect-data-platform/teradata-setup.md [hunk 1]
```diff
@@ -149,7 +149,18 @@ To learn more about dbt incremental strategies please check [the dbt incremental
...
+
+| Constraint type |	Support	Platform | enforcement |
+|-----------------|------------------|-------------|
+| not_null	      | ✅ Supported	 | ✅ Enforced |
+| primary_key	  | ✅ Supported	 | ✅ Enforced |
+| foreign_key	  | ✅ Supported	 | ✅ Enforced |
+| unique	      | ✅ Supported	 | ✅ Enforced |
+| check	          | ✅ Supported	 | ✅ Enforced |
+
+To find more on model contracts please follow dbt documentations https://docs.getdbt.com/docs/collaborate/govern/model-contracts
```
Comment (mirnawong1) [inline website/docs/docs/core/connect-data-platform/teradata-setup.md]:
```suggestion
Refer to [Model contracts](/docs/collaborate/govern/model-contracts) for more info.
```

File: website/docs/docs/core/connect-data-platform/teradata-setup.md [hunk 2]
```diff
@@ -149,15 +149,26 @@ To learn more about dbt incremental strategies please check [the dbt incremental
 All dbt commands are supported.
 
 ## Support for model contracts
-Model contracts are not yet supported with dbt-teradata.
+Model contracts are supported with dbt-teradata v1.7.1 and onwards.
+Constraint support and enforcement in dbt-teradata
```
Comment (mirnawong1) [inline website/docs/docs/core/connect-data-platform/teradata-setup.md]:
```suggestion
Constraint support and enforcement in dbt-teradata:
```

**Review by mirnawong1 — APPROVED** [review]
hey @tallamohan , thanks for opening this up! made some small tweaks and looks good to me!

**Pull request merged**